### PR TITLE
Add email portal and mailto uri

### PR DESCRIPTION
**Pull request by RyuzakiKK**
We try to use the new email portal, if it is not available we fallback to the mailto uri and as last attempt we use the xdg-email

### CODE REVIEW

**Comment by RyuzakiKK** [comment]
The two os flags that I used (O_PATH and O_CLOEXEC) are not available on Python 2.

We need to find an alternative before we can merge this PR.

File: keysign/util.py [hunk 1]
```diff
@@ -82,6 +145,41 @@ def email_file(to, from_=None, subject=None,
...
+    special path inside the sandbox. To be able to use the files from the host
+    we change the path to the absolute one. This fix in the future may not be
+    necessary because the portals should be able to automatically handle it."""
+    app_id = "org.gnome.Keysign"
+    flatpak_path = os.path.expanduser("~/.var/app/" + app_id + "/cache/tmp/")
+    fixed_files = []
+    if files:
+        for file in files:
+            if file.startswith("/var/tmp/"):
+                fixed_files.append(flatpak_path + file[9:])
```
Comment (muelli) [inline keysign/util.py]: remove the magic 9 here

Comment (RyuzakiKK) [reply keysign/util.py]: Fixed with 078845c

**Comment by muelli** [comment]
LGTM. I'm trying to build and test a flatpak with this change, but currently flatpak-builder crashes.  Let's see whether we get this sorted out.

File: keysign/util.py [hunk 2]
```diff
@@ -85,6 +149,58 @@ def email_file(to, from_=None, subject=None,
...
+    var_tmp_flat = "/var/tmp"
+    part_1 = os.path.expanduser("~/.var/app/")
+    app_id = "org.gnome.Keysign"
+    part_2 = "cache/tmp/"
+    flatpak_path = os.path.join(part_1, app_id, part_2)
+    fixed_files = []
+    if files:
+        for file in files:
+            if file.startswith(tmp_flat):
+                shutil.move(file, var_flat + file)
```
Comment (muelli) [inline keysign/util.py]: woahr. is this not a crazy hack that depends on the full file path starting with something that is also writeable in `/var`?  If, for some odd reason, the file is named "/export/tmp/foo.asc" then this construction will attempt to create `/var/export/foo.asc`, no?
This is not good.

Also, use `os.path.join`.

How about taking the basename of the file and then move (or maybe even copy only) it to `$XDG_CACHE_HOME/tmp` or something..? Where `$XDG_CACHE_HOME` should probably fall back to `~/.var/app/$ID/cache`.

**Pull request merged**